src/client.js

```diff
@@ -336,20 +337,19 @@ Client.prototype.inputLine = function(data) {
 	const irc = target.network.irc;
 	let connected = irc && irc.connection && irc.connection.connected;
 
-	if (Object.prototype.hasOwnProperty.call(inputs.userInputs, cmd) && typeof inputs.userInputs[cmd].input === "function") {
-		const plugin = inputs.userInputs[cmd];
+	if (inputs.userInputs.get(cmd) && typeof inputs.userInputs.get(cmd).input === "function") {
```

Comment:
You're calling `get` three times (the first one should have been a `has`.

Ideally you can cache `get` call into a variable, and check if it's not undefined.

Reply:
Yeah, I was thinking about that.

Reply:
Ok, I've pulled that out, but I've done it inside the if and used `has` for the if statement. Are you happy with that, or do you want it pulled out above as well? That would feel unnecessary for me, as we'd need to either do both before we know whether we have 1 or the other, or we'd have to split up the if which probably wouldn't look nice.

There's probably a nicer way to remove some of the duplication of this actually, but because they take different arguments, it would be a bit overly complicated, probably.